src/main/java/com/google/cloud/genomics/dataflow/pipelines/AlignmentsReads.java

```diff
@@ -0,0 +1,259 @@
+
+import htsjdk.samtools.util.StringUtil;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Logger;
+
+import javax.annotation.Nullable;
+
+/**
+ * Simple read counting pipeline, intended as an example for reading data from 
+ * APIs OR BAM files and invoking GATK tools.
+ */
+public class AlignmentsReads {
+  private static final Logger LOG = Logger.getLogger(AlignmentsReads.class.getName());
+  private static AlignReadsOptions options;
+  private static Pipeline p;
+  private static GenomicsFactory.OfflineAuth auth;
+  public static interface AlignReadsOptions extends GenomicsDatasetOptions, GCSOptions {
+    @Description("The ID of the Google Genomics ReadGroupSet this pipeline is working with. "
+        + "Default (empty) indicates all ReadGroupSets.")
+    @Default.String("")
+    String getReadGroupSetId();
+
+    void setReadGroupSetId(String readGroupSetId);
+
+    @Description("The path to the BAM file to get reads data from.")
+   // @Default.String("gs://genomics-public-data/platinum-genomes/bam/NA12893_S1.bam")
+    @Default.String("")
+    String getBAMFilePath();
+
+    void setBAMFilePath(String filePath);
+    
+    @Description("Whether to shard BAM reading")
+    @Default.Boolean(true)
+    boolean getShardBAMReading();
+
+    void setShardBAMReading(boolean newValue);
+    
+    
+    class ContigsFactory implements DefaultValueFactory<Iterable<Contig>> {
+      @Override
+      public Iterable<Contig> create(PipelineOptions options) {
+        return Iterables.transform(Splitter.on(",").split(options.as(AlignReadsOptions.class).getReferences()),
+            new Function<String, Contig>() {
+              @Override
+              public Contig apply(String contigString) {
+                ArrayList<String> contigInfo = newArrayList(Splitter.on(":").split(contigString));
+                return new Contig(contigInfo.get(0), 
+                    contigInfo.size() > 1 ? 
+                        Long.valueOf(contigInfo.get(1)) : 0, 
+                    contigInfo.size() > 2 ?
+                        Long.valueOf(contigInfo.get(2)) : -1);
+              }
+            });
+      }
+    }
+    
+    @Default.InstanceFactory(ContigsFactory.class)
+    @JsonIgnore
+    Iterable<Contig> getContigs();
+    
+    void setContigs(Iterable<Contig> contigs);
+  }
+  
+  public static void main(String[] args) throws GeneralSecurityException, IOException {
+    // Register the options so that they show up via --help
+    PipelineOptionsFactory.register(AlignReadsOptions.class);
+    options = PipelineOptionsFactory.fromArgs(args).withValidation().as(AlignReadsOptions.class);
+    // Option validation is not yet automatic, we make an explicit call here.
+    GenomicsDatasetOptions.Methods.validateOptions(options);
+
+    auth = GCSOptions.Methods.createGCSAuth(options);
```

Comment:
The code appears to work better if we switch to:
auth = GenomicsOptions.Methods.getGenomicsAuth(options);